collections/ansible_collections/cloudkit/service/roles/publish_templates/tasks/vms.yaml

```
@@ -0,0 +1,45 @@
+# VM Template Publishing
+- name: Get list of existing VM templates
+  ansible.builtin.uri:
+    url: "{{ publish_templates_vm_api_endpoint }}"
+    validate_certs: "{{ publish_templates_validate_certs | default(true) | bool }}"
+    headers:
+      authorization: Bearer {{ cloudkit_fulfillment_service_token }}
+  register: vm_template_check
+
+- name: Create list of existing VM template IDs
+  ansible.builtin.set_fact:
+    vm_template_ids: >-
+      {{
+      (vm_template_check.json.items | default([])) | map(attribute='id') | list
+      }}
+
+- name: Update existing VM template
+  loop: "{{ cloudkit_vm_templates }}"
+  loop_control:
+    label: "{{ template.id }}"
+    loop_var: template
+  when: vm_template_check is success and vm_template_check.json.size != 0
```

```suggestion
  when: template.id in template_ids
```
the logic has changed here, we want to update the VM templates that already exist in the API.